internal/speakerlist/speakerlist.go

```diff
@@ -71,6 +77,16 @@ func New(logger gokitlog.Logger, nodeName, bindAddr, bindPort, secret, namespace
 		sha := sha256.New()
 		mconfig.SecretKey = sha.Sum([]byte(secret))[:16]
 	}
+
+	// This channel is used by Rejoin(), that is used as a callback on node
+	// changes for k8s watch. It is assumed that operations there don't
+	// block too much time (is single thread), so make it 'big' to reduce
+	// the chances of blocking.
+	// K8s max node size is 5k, and this call back on node rejoin can be
+	// called 2-3 times (node change to Not Ready, then Ready again and
+	// sometimes another change is detected too), so we make it 5k * 3
+	sl.mlJoinCh = make(chan struct{}, 5000*3)
```

Comment:
len should be 1, and to queue something:
```
select {
case sl.mlJoinCh <- struct{}{}:
    // some log
default:
    // some other log
}
```

Reply:
>  This is worse than that, this is a split brain and you might have multiple nodes responding to ARP requests causing the tcp connection to switch nodes and get tcp reset

Yes, true. Didn't go into the details as I linked the issue. Tried to explain but the commit message was already big and explaining that in more detail was complicated while being clear, so I just explained that bit and let the link to the issue do the rest :)

Reply:
`sl.mlJoinCh = make(chan struct{}, 1)`
you understand me correctly ;)

Reply:
Fixed.